FabricObserver.Extensibility/ObserverBase.cs

```diff
@@ -1140,27 +1154,44 @@ public bool DumpWindowsServiceProcess(int processId, string procName, string met
                         errorWarningCode = (healthState == HealthState.Error) ?
                             FOErrorWarningCodes.NodeErrorTotalOpenFileHandlesPercent : FOErrorWarningCodes.NodeWarningTotalOpenFileHandlesPercent;
                         break;
+
+                    // Process Private Bytes.
+                    case ErrorWarningProperty.PrivateBytesMb when entityType == EntityType.Application || entityType == EntityType.Service:
+                        errorWarningCode = (healthState == HealthState.Error) ?
+                            FOErrorWarningCodes.AppErrorPrivateBytesMb : FOErrorWarningCodes.AppWarningPrivateBytesMb;
+                        break;
+
+                    // RG Memory Limit Percent. Only Warning is supported.
+                    case ErrorWarningProperty.RGMemoryUsagePercent when entityType == EntityType.Application || entityType == EntityType.Service:
+                        errorWarningCode = FOErrorWarningCodes.AppWarningRGMemoryLimitPercent;
+                        break;
                 }
 
                 var healthMessage = new StringBuilder();
                 string childProcMsg = string.Empty;
+                string rgInfo = string.Empty;
+                string drive = string.Empty;
 
                 if (replicaOrInstance != null && replicaOrInstance.ChildProcesses != null)
                 {
                     childProcMsg = $" Note that {serviceName.OriginalString} has spawned one or more child processes ({replicaOrInstance.ChildProcesses.Count}). " +
                                    $"Their cumulative impact on {processName}'s resource usage has been applied.";
                 }
 
-                string drive = string.Empty;
+                if (replicaOrInstance != null && data.Property == ErrorWarningProperty.RGMemoryUsagePercent)
```

Comment:
will we log the RGMEmoryLimitMb in case we have a code package without replicas being hosted?